pkg/build/interfaces.go

```diff
@@ -41,11 +41,20 @@ type Downloader interface {
 	Download(*api.Config) (*api.SourceInfo, error)
 }
 
+// Implement ignore file type processing on source tree
+// NOTE: raised to this level for possible future extensions to
+// support say both .gitignore and .dockerignore level functionality
+// ( currently do .dockerignore)
```

Comment:
ah hah. i should have read ahead before commenting.  I still think this bears some restructuring but it's not so far off from what i described.